serde_derive/src/internals/check.rs

```diff
@@ -278,6 +279,31 @@ fn check_adjacent_tag_conflict(cx: &Ctxt, cont: &Container) {
     }
 }
 
+/// Enums and unit structs cannot be transparent.
```

Comment:
Technically single variant enums are just structs

Reply:
True but let's leave that until someone asks for it.